### PR TITLE
Adding new syntax tests

### PR DESCRIPTION
**Pull request by CelsoSimoes**
New regression tests; attempt to find edge cases.

### CODE REVIEW

File: src/test/fixtures/syntax-error/error-1/Main.hp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Syntax; 
+// Esse teste PASSA. (Baseado no WrongName)
```
Comment (flavluc) [inline src/test/fixtures/syntax-error/error-1/Main.hp]: remove the comments

File: src/test/kotlin/VisitorTest.kt [hunk 2]
```diff
@@ -36,4 +36,48 @@ class VisitorTest {
       assertEquals("undefined name: WrongName::bob", exception.message)
 
     }
+
+    @Test
+    @DisplayName("Should return a null pointer Error") // kotlin.KotlinNullPointerException
```
Comment (flavluc) [inline src/test/kotlin/VisitorTest.kt]: Give it a more descriptive name. What is this test case testing?

File: src/test/kotlin/VisitorTest.kt [hunk 3]
```diff
@@ -36,4 +36,48 @@ class VisitorTest {
...
+    fun shouldReturnSyntaxError1() {
+      val file = "src/test/fixtures/syntax-error/error-1/Main.hp"
+      val priority = listOf("Actors", "Actions", "Resources")
+      val exception = assertFailsWith<Exception> { IR.generate(file, priority) }
+      assertEquals(null, exception.message)
```
Comment (flavluc) [inline src/test/kotlin/VisitorTest.kt]: Errors should be handled with a descriptive message instead of a not cool `KotlinNullPointerException`. What would be a good message that make explicit the error that is occurring?

File: src/test/kotlin/VisitorTest.kt [hunk 4]
```diff
@@ -36,4 +36,48 @@ class VisitorTest {
...
+    }
+
+    @Test
+    @DisplayName("Should return a Syntax Error") // Or Should DENY Alice Reads CCN
+    fun shouldReturnSyntaxError2() {
+      val file = "src/test/fixtures/syntax-error/error-2/Main.hp"
+      val priority = listOf("Actors", "Actions", "Resources")
+      //val exception = assertFailsWith<Exception> { IR.generate(file, priority) }
```
Comment (flavluc) [inline src/test/kotlin/VisitorTest.kt]: remove comments

File: src/test/kotlin/VisitorTest.kt [hunk 5]
```diff
@@ -36,4 +36,48 @@ class VisitorTest {
...
+      //assertEquals("Some syntax error", exception.message)
+      val ir = (IR.generate(file, priority) as IRNode).ir
+      val expectedIRString = "{Bob={Updates=[SSN, EMAIL, CCN], Deletes=[SSN, EMAIL, CCN], Reads=[SSN, EMAIL, CCN]}, Alice={Updates=[SSN, EMAIL, CCN], Deletes=[SSN, EMAIL, CCN], Reads=[SSN, EMAIL]}}"
+      assertThat(ir.toString()).isEqualTo(expectedIRString)
+
+    }
+
+    @Test
+    @DisplayName("Should DENY Alice Reads CCN") // Or Should return a Syntax Error
```
Comment (flavluc) [inline src/test/kotlin/VisitorTest.kt]: Is this testing a syntax error? which one?

**Review by flavluc — COMMENTED** [review]
@CelsoSimoes The PR looks good, but there are some changes that need to be done. Mostly removing comments and commented code.

Also, a person should be able to tell what is being tested just by looking at the test case. What is `error-1` inside `syntax-error` folder? Can you give it a more descriptive name? What about the others?

**Comment by CelsoSimoes** [comment]
Done:
- Refactored tests, with folder changes to try to fit a theme better;
- Comments removed;
- Suggestive Names added;

What's next:
 - Make the 2 failed checks pass. For this:
   - Add a 'more cool' message for a `KotlinNullPointerException` instead of just `<null>`;
   - Define what kind of error the program should appoint when theres an DENY after an ALLOW on imported variables;
      exemplified on Test "Should return a syntax error on imported variable: DENY after an ALLOW".
